_posts/2017-10-06-rust-blog-series-2.md

```diff
@@ -0,0 +1,1144 @@
+but it's also very easy to get started with,
+which is why we're using it here.
+
+To enable this feature, we must create the file `src/schema.rs`
+
+```rust
+// Inside `src/schema.rs`
+
+infer_schema!("dotenv:DATABASE_URL");
+```
+
+This file will be brought into scope in our `src/lib.rs` file  via a `use` statement.
+Doing so will enable access to our schema from anywhere in our app.
+
+[schema in depth guide]: https://github.com/diesel-rs/diesel/blob/master/guide_drafts/schema-in-depth.md
+
+
+## Database Models
+
+Now that we have a schema, Rust can interact with database tables,
+however that is a little different than interacting with a database model
+(referring to the "M" of the [MVC] pattern I'm used to in Ruby on Rails).
+Diesel is a bit different than other ORMS, in that its main focus is on query building.
+I expected to work with a single object as my database model,
+however in Diesel you should feel comfortable defining specialized structs to support your queries.
+This may seem odd at first, but it should feel more natural by the time we get to designing all CRUD endpoints.
+
+[MVC]: https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller#Components
+
+At the moment, we really only care about two database operations; Create and Read.
+This amounts to SQL INSERT and SELECT statements.
+
+We have two database tables, users and posts, and we need to perform two operations on each.
+The data structs for our models *may* look like this.
+- User
+- NewUser
+- Post
+- NewPost
+
+Lets create a `src/models.rs` file and review the code that goes into it.
+
+
+```rust
+// Inside `src/models.rs`
+
+// This `models` file will also be imported into our `lib`
+// We JUST made the schema file...
+// Lets take advantage of it by bringing it into scope here
+use schema::{posts, users};
+
+#[derive(Debug, Queryable)]
+struct User {
+    id: i32,
+    first_name: String,
+    last_name: String,
+    email: String,
+    password: String, 
+}
+
+#[derive(Debug, Insertable)]
+#[table_name="users"]
+struct NewUser {
+    first_name: String,
+    last_name: String,
+    email: String,
+    password: String, 
+}
+
+#[derive(Debug, Queryable)]
+struct Post {
+    id: i32,
+    user_id: i32,
+    title: String,
+    content: String,
+    published: bool,
+}
+
+#[derive(Debug, Insertable)]
+#[table_name="posts"]
+struct NewPost {
+    user_id: i32,
+    title: String,
+    content: String,
+}
+
+```
+
+If you're thinking *"whoa, whoa, whoa... WUT are those weird DERIVE things?!?!"*,
+don't worry, they're your friend!
+
+The `diesel_codegen` crate takes all of those `#[derive()]`'s
+and writes boilerplate code so we may focus on the business logic of our application.
+
+`Debug` should look somewhat familiar.
+It enables you to print/format values that otherwise don't have a way to be printed.
+If you're unfamiliar, please check out the official [Derive] docs.
+
+The really interesting derives are `Queryable` and `Insertable`.
```

Comment:
Either change that to `Insertable` and `Queryable` or invert them below. I was really confused about `Insertable` being first.